### PR TITLE
[VL] Use Velox's monolithic build

### PR DESCRIPTION
**Pull request by PHILO-HE**
## What changes were proposed in this pull request?

Monolithic build was introduced into velox by https://github.com/facebookincubator/velox/pull/9948. It will produce a unified velox lib, libvelox.a. With it, linking velox libs in Gluten becomes less complex and easier to maintain.

We also found Velox's test util libs are still separate. They are required when Gluten --build_tests=ON. In this case, we still keep linking these separate libs.

## How was this patch tested?

CI.

### CODE REVIEW

File: cpp/velox/CMakeLists.txt [hunk 1]
```diff
@@ -582,9 +265,71 @@ if(BUILD_JEMALLOC)
...
 target_link_libraries(velox PUBLIC gluten)
-add_velox_dependencies()
+
+add_library(facebook::velox STATIC IMPORTED)
+set_target_properties(
+  facebook::velox PROPERTIES IMPORTED_LOCATION
+                             ${VELOX_BUILD_PATH}/lib/libvelox.a)
+
+if(BUILD_TESTS)
+  add_library(facebook::velox::dbgen STATIC IMPORTED)
```
Comment (xumingming) [inline cpp/velox/CMakeLists.txt:275]: @PHILO-HE For the testing libraries, they are still separate libraries, right?

File: cpp/velox/CMakeLists.txt [hunk 2]
```diff
@@ -582,9 +265,71 @@ if(BUILD_JEMALLOC)
...
+    facebook::velox::vector_test
+    PROPERTIES
+      IMPORTED_LOCATION
+      "${VELOX_BUILD_PATH}/velox/vector/tests/utils/libvelox_vector_test_lib.a")
+
+  add_library(facebook::velox::dwio_common_test STATIC IMPORTED)
+  set_target_properties(
+    facebook::velox::dwio_common_test
+    PROPERTIES
+      IMPORTED_LOCATION
```
Comment (rui-mo) [inline cpp/velox/CMakeLists.txt:293]: Please consider using a helper function to simply the code.

**Review by rui-mo — COMMENTED** [review]
Thanks!

File: cpp/velox/CMakeLists.txt (hunk 1 above)
Comment (PHILO-HE) [reply cpp/velox/CMakeLists.txt:276]: @xumingming, yes, these test libs are still separate.

**Comment by gaoyangxiaozhu** [comment]
cool feature related to build!

**Comment by PHILO-HE** [comment]
@zhztheplayer, @xumingming, do you have any comments?
I checked the overall size of all separate velox libs when monolithic build is off. It roughly equals to the size of libvelox.a.

**Review by zhztheplayer — APPROVED** [review]
Thanks for the effort!

**Pull request merged**